### PR TITLE
chore: update controller-runtime.

### PR DESCRIPTION
**Pull request by jvanz**
## Description

Updates the controller-runtime and kubewarden-controller package.

### CODE REVIEW

**Review by viccuad — APPROVED** [review]
LGTM, waiting for controller.

File: internal/report/store_test.go
Comment (flavio) [inline internal/report/store_test.go:222]: maybe not related, why do we have the `nolint` there? was that because fake client was temporarily marked as deprecated (hence a linter error)?

Can we remove the `// nolint` directive or make it more specific (as in, make explicit which linter check is disabled)?

This comment applies to some other parts of this file

**Review by flavio — APPROVED** [review]
Overall LGTM, I just left a minor comment

**Review by fabriziosestito — APPROVED** [review]
LGTM when the nolint comment is addressed

**Comment by jvanz** [comment]
This is blocked until we release the controller with the latest updates

**Comment by viccuad** [comment]
Bumped kubewarden-controller dep to v1.7.1, tested locally, all fine.

**Comment by viccuad** [comment]
Merging, CI is fine. Many thanks José!

**Pull request merged**
